runtime/lua/vim/ui.lua

```diff
@@ -104,4 +104,49 @@ function M.input(opts, on_confirm)
   end
 end
 
+--- Opens a URI by sending it to the system's default handler.
+--- This function utilizes `xdg-open`, `wslview` / `start`, or `open` commands
+--- depending on the system to open the provided URI.
+---
+---
+---@param uri string Uri to be opened
+---
+---@return string returns the string output of the command. See |system()|.
+---
+--- Example:
+--- <pre>lua
+--- vim.ui.open_uri("https://neovim.io/")
+---
+--- vim.ui.open_uri("/path/to/file")
+--- </pre>
+function M.open_uri(uri)
```

Comment:
Perhaps we should just name it `vim.ui.open`, mentioning "URI" in the name is perhaps confusing because it also supported non-URI filesystem names. 

Opening stuff in the editor (Nvim) is called "edit" rather than "open" so that hopefully is enough distinction.

Reply:
I think os_open is a good fit